src/servers/src/http.rs

```diff
@@ -239,14 +240,21 @@ impl HttpRecordsOutput {
         } else {
             let num_rows = recordbatches.iter().map(|r| r.num_rows()).sum::<usize>();
             let mut rows = Vec::with_capacity(num_rows);
+            let schemas = schema.column_schemas();
             let num_cols = schema.column_schemas().len();
             rows.resize_with(num_rows, || Vec::with_capacity(num_cols));
 
             let mut finished_row_cursor = 0;
             for recordbatch in recordbatches {
-                for col in recordbatch.columns() {
+                for (col_idx, col) in recordbatch.columns().iter().enumerate() {
+                    // safety here: schemas length is equal to the number of columns in the recordbatch
+                    let schema = &schemas[col_idx];
                     for row_idx in 0..recordbatch.num_rows() {
-                        let value = Value::try_from(col.get_ref(row_idx)).context(ToJsonSnafu)?;
+                        let column_pair = ColumnPair {
```

Comment:
The whole purpose of `ColumnPair` seems only to help convert JSON data. Maybe we can just use a helper function instead of introducing `ColumnPair` and use `From`.